packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -388,37 +388,45 @@ function getSchemaForUnion(
   const variants = Array.from(union.variants.values());
 
   let values = [];
-
+  let namedUnionMember = false;
   for (const variant of variants) {
     // We already know it's not a model type
-    values.push(
-      getSchemaForType(dpgContext, variant.type, { ...options, needRef: false })
-    );
+    const variantType = getSchemaForType(dpgContext, variant.type, {
+      ...options,
+      needRef: false
+    });
+    values.push(variantType);
+    if (variantType.typeName) {
+      namedUnionMember = true;
+    }
   }
-  if (asEnum?.open) {
+  if (asEnum?.open && asEnum.members.size > 0) {
+    namedUnionMember = false;
     values = [];
     for (const [_, member] of asEnum.members.entries()) {
-      values.push(
-        getSchemaForType(dpgContext, member.type, {
-          ...options,
-          needRef: false
-        })
-      );
+      const memberType = getSchemaForType(dpgContext, member.type, {
+        ...options,
+        needRef: false
+      });
+      values.push(memberType);
+      if (memberType.name) {
+        namedUnionMember = true;
+      }
     }
   }
   const schema: any = {};
   if (values.length > 0) {
     schema.enum = values;
     const unionAlias =
-      asEnum?.open && asEnum?.kind
+      asEnum?.open && asEnum?.kind && !namedUnionMember
```

Comment:
could you help me understand why the enum member having a name would impact the union alias?

Reply:
```
      union ExtensibleStringEnum {
        string;
        "a";
        "b";
      }
     model SimpleModel {
        propNullableRequired: ExtensibleStringEnum | null;
        propNullableFlattened: string | "a" | "b" | null;
     }
```
In the case, `getUnionAsEnum` for `propNullableRequired` and `propNullableFlattened` type will be a valid enum, which means `asEnum.open` is true and `asEnum.kind` has meaningful value. but we can't generate string type for `propNullableRequired` and `propNullableFlattened`. 
```
       export interface SimpleModel {
         propNullableRequired: ExtensibleStringEnum | null;
         propNullableFlattened: string | null;
       }
```
in the `propNullableRequired` case, we should get each value type, therefore, we need to see if the there's any named union in the member. 
in the `propNullableFlattened` case, we should leverage the `asEnum.kind` and `asEnum.nullable` to get the right typescript type.

Reply:
maybe not a priority, but we may meet case for mixing both.

```
model SimpleModel {
        propNullableRequired: ExtensibleStringEnum | string | "a" | null;
}
```

Reply:
`string | null` is our answer here.  but I think we should avoid writing typespec like that.